typing/typecore.ml

```diff
@@ -4244,18 +4454,23 @@ and type_expect_
         match expected_mode.position with
         | RNontail ->
           raise (Error (loc, env, Exclave_in_nontail_position))
-        | RTail (mode, _) ->
+        | RTail (regionality, _) ->
           (* mode' is RNontail, because currently our language cannot construct
              region in the tail of another region.*)
-          let mode' = mode_default mode in
+          let mode' =
+            mode_default
+              (Value.with_locality regionality expected_mode.mode)
+          in
           (* The middle-end relies on all functions which allocate into their
              parent's region having a return mode of local. *)
-          submode ~loc ~env ~reason:Other Value_mode.local mode';
+          submode ~loc ~env ~reason:Other
+            (Value.min_with_locality Regionality.local) mode';
```

Comment:
With the new solver I think we should have submodules for e.g. `Value.Left` and `Value.Right`. Then we can have things like `Value.Left.local` for `Value.min_with_locality Regionality.local`.

Reply:
Yeah that sounds like a good idea. But I think `Value.local` is sufficient, because "being left" should be encoded in the types already.